### PR TITLE
Backporting Harbor fixes to master

### PR DESCRIPTION
**Pull request by cdchris12**
# Checklist

- [ ] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated
- [X] PR title is ready for changelog and subsystem label(s) applied

This PR fixes the issue with the HarborSync script not working after the migration to Harbor v2.1.x

### CODE REVIEW

File: services/api/src/resources/project/harborSetup.ts [hunk 1]
```diff
@@ -226,8 +226,8 @@ async function resetHarborWebhook(sqlClient: MariaClient, harborClient, lagoonPr
           }
         ],
         event_types: [
```
Comment (tobybellwood) [inline services/api/src/resources/project/harborSetup.ts]: my one question here is whether we need to wrap this in a HARBOR_API_VERSION variable check to make it compatible with Harbor v1?

Comment (cdchris12) [reply services/api/src/resources/project/harborSetup.ts]: Good catch. I'll write that in

Comment (tobybellwood) [reply services/api/src/resources/project/harborSetup.ts]: giphy-downsized

**Comment by tobybellwood** [comment]
The test failure here is real
building locally with `HARBOR_API_VERSION=` set to an empty string in docker-compose is somehow triggering it to mangle the API URL in harborclient
```
The push refers to repository [172.16.0.1:8084/ci-api-control-k8s/pr-1/node]
Get http://172.16.0.1:8084/v2/: dial tcp 172.16.0.1:8084: connect: connection refused
```

**Pull request merged**